### PR TITLE
fix: Set basic_auth to optional in self_managed_kafka_parameters

### LINKED ISSUE #33025: [Bug]: Incorrect validation of 'basic_auth' in resource 'aws_pipes_pipe' for self managed Kafka configuration
### Terraform Core Version

1.5.5

### AWS Provider Version

5.12.0

### Affected Resource(s)

* `aws_pipes_pipe`, section: `source_parameters.self_managed_kafka_parameters.credentials.basic_auth`

### Expected Behavior

When credentials for a Self Managed Kafka source are configured, the "basic_auth" property should be optional, so that it is not required when configuring TLS or SASL credentials.

### Actual Behavior

Validation fails when the "basic_auth" configuration is omitted.

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```tf
resource "aws_pipes_pipe" "example" {
  name     = "example-pipe"
  role_arn = var.role_arn
  source   = var.source
  target   = var.target

  source_parameters {
    self_managed_kafka_parameters {
      additional_bootstrap_servers = [ var.aditional_sever_1, var.aditional_sever_2 ]
      credentials {
        client_certificate_tls_auth = var.client_certificate_tls_auth
      }
      topic_name = var.topic_name
    }
  }
}
```

### Steps to Reproduce

* Configure a new pipe using a Self Managed Kafka as source, with credentials other than "basic_auth". 
* When running `terraform validate`, the following error will be displayed:
```
The argument "basic_auth" is required, but no definition was found.
```

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

The error is in the schema definition, in the following line of code:
https://github.com/hashicorp/terraform-provider-aws/blob/e8931004742d138892921e19456b0b98ae9292a0/internal/service/pipes/source_parameters.go#L447
Since it is defined as `Required`, the "basic_auth" setting is mandatory.

### References

_No response_

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by vzilinas**
### Description

Fixes the validation on aws_pipes_pipe.source_parameters.self_managed_kafka_parameters.credentials to make basic_auth not required. This is already correct in documentation https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/pipes_pipe#self_managed_kafka_parameters

### Relations
Closes #33025

### Output from Acceptance Testing

```console
```

### CODE REVIEW

**Review by extmaciejturek — APPROVED** [review]

File: internal/service/pipes/source_parameters.go [hunk 1]
```diff
@@ -444,7 +444,7 @@ func sourceParametersSchema() *schema.Schema {
 									Schema: map[string]*schema.Schema{
 										"basic_auth": {
 											Type:         schema.TypeString,
-											Required:     true,
+											Optional:     true,
```
Comment (b4ry) [inline internal/service/pipes/source_parameters.go:447]: what would setting required to false do? the same i guess?

**Review by b4ry — APPROVED** [review]

File: internal/service/pipes/source_parameters.go (hunk 1 above)
Comment (vzilinas) [reply internal/service/pipes/source_parameters.go:447]: I have decided to stick with conventions based on the naming below

**Review by petercsepkeEF — APPROVED** [review]

**Review by djanowicz — APPROVED** [review]

**Review by miki-albert — APPROVED** [review]

**Review by marek-kwitek-ef — APPROVED** [review]
Looks good.

**Comment by erlandsen-tech** [comment]
Is this not merged, as it is still an issue.

**Comment by abenhadjhammouda** [comment]
@justinretzolk  @ewbankkit Is it possible to merge this PR please ? It is blocking for us. Thanks

**Comment by jgaynon** [comment]
This is still an issue in provider version 5.52.0. 

This **1 line** PR fixes a bug https://github.com/hashicorp/terraform-provider-aws/issues/33025 open for 10 months and yet it's been sitting, approved but not merged, for 6 months? This PR really needs to be merged, the bug completely blocks creating an EventBridge pipe with a self managed Kafka source that doesn't use basic auth - any serious production usage of Kafka will not be using basic auth.

**Comment by erlandsen-tech** [comment]
We are using a workaround where we deploy the pipe and then via cli we add authentication.. Any feedback on this would be appreciated from our side as well. Kind regards, John Inge Erlandsen5. juni 2024 kl. 15:52 skrev jgaynon ***@***.***>:﻿
This is still an issue in provider version 5.52.0.
This 1 line PR fixes a bug #33025 open for 10 months and yet it's been sitting, approved but not merged, for 6 months? This PR really needs to be merged, the bug completely blocks creating an EventBridge pipe with a self managed Kafka source that doesn't use basic auth - any serious production usage of Kafka will not be using basic auth.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccPipesPipe_selfManagedKafkaSourceLambdaFunctionTarget' PKG=pipes
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/pipes/... -v -count 1 -parallel 20  -run=TestAccPipesPipe_selfManagedKafkaSourceLambdaFunctionTarget -timeout 360m
=== RUN   TestAccPipesPipe_selfManagedKafkaSourceLambdaFunctionTarget
=== PAUSE TestAccPipesPipe_selfManagedKafkaSourceLambdaFunctionTarget
=== CONT  TestAccPipesPipe_selfManagedKafkaSourceLambdaFunctionTarget
    pipe_test.go:1126: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Security Group (sg-0871bf5a0727f5338): DependencyViolation: resource sg-0871bf5a0727f5338 has a dependent object
        	status code: 400, request id: 6250a9d0-3d5a-4ed0-8a63-9c8581299485
        
        
        Error: deleting EC2 Subnet (subnet-0a9de4f3eb3645acf): DependencyViolation: The subnet 'subnet-0a9de4f3eb3645acf' has dependencies and cannot be deleted.
        	status code: 400, request id: 45977352-1d7e-46e2-8c5e-a7034fb355ee
        
        
        Error: deleting EC2 Subnet (subnet-0c60e15176cb5723d): DependencyViolation: The subnet 'subnet-0c60e15176cb5723d' has dependencies and cannot be deleted.
        	status code: 400, request id: 365924a4-dd38-44df-9d28-a9d256081c4c
        
--- FAIL: TestAccPipesPipe_selfManagedKafkaSourceLambdaFunctionTarget (1460.73s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/pipes	1465.338s
FAIL
make: *** [testacc] Error 1
```

Errors are expected.

**Comment by ewbankkit** [comment]
@vzilinas Thanks for the contribution :tada: :clap:.

**Pull request merged**